src/repo.rs

```diff
@@ -1840,6 +1841,77 @@ impl Repository {
         Ok(())
     }
 
+    /// Merge two files as they exist in the in-memory data structures, using
+    /// the given common ancestor as the baseline, producing a
+    /// `git_merge_file_result` that reflects the merge result.  The
+    /// `git_merge_file_result` must be freed with `git_merge_file_result_free`.
+    ///
+    /// Note that this function does not reference a repository and any
+    /// configuration must be passed as `git_merge_file_options`.
+    ///
+    /// @param out The git_merge_file_result to be filled in
+    /// @param ancestor The contents of the ancestor file
+    /// @param ours The contents of the file in "our" side
+    /// @param theirs The contents of the file in "their" side
+    /// @param opts The merge file options or `NULL` for defaults
+    /// @return 0 on success or error code
+    pub fn merge_file(
+        &self,
+        ancestor: Option<&IndexEntry>,
+        ours: Option<&IndexEntry>,
+        theirs: Option<&IndexEntry>,
+        options: Option<&MergeFileOptions>,
+    ) -> Result<MergeFileResult, Error> {
+        let ancestor_input;
+        let ours_input;
+        let theirs_input;
+
+        let ancestor_raw;
+        let ours_raw;
+        let theirs_raw;
+
+        if let Some(ancestor) = ancestor {
+            ancestor_input = MergeFileInput::from(&self, ancestor);
+            ancestor_raw = ancestor_input.raw();
+        } else {
+            ancestor_raw = ptr::null();
+        }
+        if let Some(ours) = ours {
+            ours_input = MergeFileInput::from(&self, ours);
+            ours_raw = ours_input.raw();
+        } else {
+            ours_raw = ptr::null();
+        }
+        if let Some(theirs) = theirs {
+            theirs_input = MergeFileInput::from(&self, theirs);
```

Comment:
FWIW I think it would be best to take `theirs: Option<&MergeFileInput>` in this function instead of doing this computation internally.

Reply:
Understood, now it accepts `Option<&MergeFileInput>`